src/StatsdClient/Aggregator/SetAggregator.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections.Generic;
+using StatsdClient.Statistic;
+using StatsdClient.Utils;
+
+namespace StatsdClient.Aggregator
+{
+    /// <summary>
+    /// Aggregate `StatsMetric` instances of type `Set` by keeping the unique values.
+    /// </summary>
+    internal class SetAggregator
+    {
+        private readonly AggregatorFlusher<StatsMetricSet> _aggregator;
+        private readonly Pool<StatsMetricSet> _pool;
+        private readonly Telemetry _optionalTelemetry;
+
+        public SetAggregator(MetricAggregatorParameters parameters, Telemetry optionalTelemetry)
+        {
+            _aggregator = new AggregatorFlusher<StatsMetricSet>(parameters, MetricType.Set);
+            _pool = new Pool<StatsMetricSet>(pool => new StatsMetricSet(pool), 2 * parameters.MaxUniqueStatsBeforeFlush);
```

Comment:
`CountAggregator` does not allocate memory as it sum integers. 
`SetAggregator` allocates a `HashSet<string>` by metric. I tried to avoid memory allocations. As I already have a `Pool` it does not introduce too much complexity. What do you think?